main.libsonnet

```diff
@@ -13,4 +13,35 @@ local d = import 'doc-util/main.libsonnet';
   ),
 
   url: (import './url.libsonnet'),
+
+  camelcase: (import './camelcase.libsonnet'),
+
+  local cp(c) = std.codepoint(c),
+
+  '#isAsciiLower':: d.fn(
+    '`isAsciiLower` reports whether ASCII character `c` is a lower case letter',
+    [d.arg('c', d.T.string)]
+  ),
+  isAsciiLower(c):
+    if cp(c) >= 97 && cp(c) < 123
+    then true
+    else false,
+
+  '#isAsciiUpper':: d.fn(
+    '`isAsciiUpper` reports whether ASCII character `c` is a upper case letter',
+    [d.arg('c', d.T.string)]
+  ),
+  isAsciiUpper(c):
+    if cp(c) >= 65 && cp(c) < 91
+    then true
+    else false,
+
+  '#isNumber':: d.fn(
+    '`isNumber` reports whether character `c` is a number.',
+    [d.arg('c', d.T.string)]
+  ),
+  isNumber(c):
+    if std.isNumber(c) || (cp(c) >= 48 && cp(c) < 58)
+    then true
+    else false,
```

Comment:
Nice, but let's not put anything in the `xtd` root namespace, that might get out of hand quickly.

What about `ascii.isLower`, etc. Similar to the Go[ `unicode`](https://pkg.go.dev/unicode) package?

Reply:
That sounds reasonable, will implement.